### PR TITLE
Upstreaming a few upgrades

### PR DESCRIPTION
**Pull request by lostmsu**
See individual commits.

`CollectionElementInitializer` action was removed because the new version of Roslyn for C# 9 treats collection initializers as `Add` method invocations, so a separate handler is not needed.

### CODE REVIEW

File: src/NonCopyable/NonCopyable/NonCopyableAnalyzer.cs [hunk 1]
```diff
@@ -101,18 +101,6 @@ public override void Initialize(AnalysisContext context)
...
-                    var op = (ICollectionElementInitializerOperation)oc.Operation;
-
-                    if (!HasNonCopyableParameter(op.AddMethod)) return;
-
-                    foreach (var a in op.Arguments)
-                    {
-                        CheckCopyability(oc, a, InitializerRule);
-                    }
-                }, OperationKind.CollectionElementInitializer);
-
```
Comment (ufcpp) [inline src/NonCopyable/NonCopyable/NonCopyableAnalyzer.cs:115]: Is this change related to supporting C# 9?

File: src/NonCopyable/NonCopyable/NonCopyableAnalyzer.cs [hunk 2]
```diff
@@ -61,6 +61,18 @@ public override void Initialize(AnalysisContext context)
...
+                    // return of a local variable or a parameter as a "move".
+                    if ((op.ReturnedValue.Kind == OperationKind.LocalReference
+                        || op.ReturnedValue.Kind == OperationKind.ParameterReference)
+                        && op.Kind == OperationKind.Return)
+                    {
+                        var varScope = op.ReturnedValue.GetSymbol().ContainingSymbol;
+                        var opScope = op.SemanticModel.GetEnclosingSymbol(op.Syntax.SpanStart);
+                        if (SymbolEqualityComparer.Default.Equals(varScope, opScope)) return;
+                    }
+
```
Comment (ufcpp) [inline src/NonCopyable/NonCopyable/NonCopyableAnalyzer.cs:75]: While I understand the demand, should this be included in this pull request?

File: src/NonCopyable/NonCopyable/NonCopyableAnalyzer.cs (hunk 1 above)
Comment (lostmsu) [reply src/NonCopyable/NonCopyable/NonCopyableAnalyzer.cs:115]: Yeah, it is in the comment about this PR:

> `CollectionElementInitializer` action was removed because the new version of Roslyn for C# 9 treats collection initializers as `Add` method invocations, so a separate handler is not needed.

File: src/NonCopyable/NonCopyable/NonCopyableAnalyzer.cs (hunk 2 above)
Comment (lostmsu) [reply src/NonCopyable/NonCopyable/NonCopyableAnalyzer.cs:75]: @ufcpp TBH, I was too lazy to split the changes in multiple PRs. E.g. this is not a PR to support C# 9, this is a PR to bring all the changes I made to enhance NonCopyableAnalyzer for our open source project https://github.com/pythonnet/pythonnet/ in one swat.

If you really insist, I can make separate PRs, but it should be easy enough to review individual commits in this one.

Comment (ufcpp) [reply src/NonCopyable/NonCopyable/NonCopyableAnalyzer.cs:75]: OK. Don't mind. I'll merge this PR and release the package in my spare time.

Comment (ufcpp) [reply src/NonCopyable/NonCopyable/NonCopyableAnalyzer.cs:75]: https://www.nuget.org/packages/NonCopyableAnalyzer/0.7.0
